### PR TITLE
test(vertical-menu): playwright refactor

### PR DESCRIPTION
**Pull request by stephenogorman**
### Proposed behaviour

Refactor Vertical-Menu tests from Cypress to Playwright

### Current behaviour

Tests are currently Cypress

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Playwright automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [ ] Related docs have been updated if required

#### QA

- [x] Tested in CodeSandbox/storybook
- [x] Add new Playwright test coverage if required
- [x] Carbon implementation matches Design System/designs
- [x] UI Tests GitHub check reviewed if required

### Additional context

N/A

### Testing instructions
- [x] Run Vertical Menu tests in Playwright Test Runner to check if the `vertical-menu.pw.tsx` file passed
- [x] Run all other tests in Playwright Test Runner to check none of the other `*.pw.tsx` files have regressed

### CODE REVIEW

File: src/components/vertical-menu/components.test-pw.tsx [hunk 1]
```diff
@@ -0,0 +1,382 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
```
Comment (DipperTheDan) [inline src/components/vertical-menu/components.test-pw.tsx:1]: question(non-blocking): Do we still need this comment?

File: src/components/vertical-menu/components.test-pw.tsx [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+          <VerticalMenuItem title="ChildItem 1" href="/child-item-1" />
+          <VerticalMenuItem title="ChildItem 2" href="/child-item-2" />
+        </VerticalMenuItem>
+      </VerticalMenu>
+    </Box>
+  );
+};
+
+export const VerticalMenuItemCustom = ({
+  component,
```
Comment (DipperTheDan) [inline src/components/vertical-menu/components.test-pw.tsx:103]: suggestion(non-blocking): I don't think you need this as I cannot see where it's used in this example.

File: src/components/vertical-menu/vertical-menu.pw.tsx [hunk 3]
```diff
@@ -0,0 +1,702 @@
...
+    await expect(pill1).toHaveAttribute("data-component", "pill");
+    await expect(pill1).toBeVisible();
+    await expect(pill2).toHaveCount(0);
+  });
+
+  test(`should render with iconType prop`, async ({ mount, page }) => {
+    await mount(<VerticalMenuItemCustom iconType="filter" />);
+
+    const filter = verticalMenuItem(page).first().locator("span").first();
+    await expect(filter).toHaveAttribute("data-element", "filter");
```
Comment (DipperTheDan) [inline src/components/vertical-menu/vertical-menu.pw.tsx:132]:
```suggestion
    const filter = verticalMenuItem(page).first().locator("span").first();
    
    await expect(filter).toHaveAttribute("data-element", "filter");
```

File: src/components/vertical-menu/vertical-menu.pw.tsx [hunk 4]
```diff
@@ -0,0 +1,702 @@
...
+      const menuItem = document.querySelector(
+        `[data-component="vertical-menu-item"]`
+      );
+      if (!menuItem) {
+        return null;
+      }
+      const beforePseudoElement = window.getComputedStyle(menuItem, "::before");
+      return beforePseudoElement ? beforePseudoElement.borderRadius : null;
+    });
+    expect(borderRadius).toBe("8px");
```
Comment (DipperTheDan) [inline src/components/vertical-menu/vertical-menu.pw.tsx:176]:
```suggestion
    });
    
    expect(borderRadius).toBe("8px");
```

File: src/components/vertical-menu/vertical-menu.pw.tsx [hunk 5]
```diff
@@ -0,0 +1,702 @@
...
+    page,
+  }) => {
+    await mount(<VerticalMenuDefaultComponent />);
+
+    await verticalMenuItem(page).first().focus();
+    await page.keyboard.press("Tab");
+    await page.keyboard.press("Tab");
+    await page.keyboard.press("Tab");
+    await page.keyboard.press("Tab");
+    await expect(
```
Comment (DipperTheDan) [inline src/components/vertical-menu/vertical-menu.pw.tsx:390]:
```suggestion
    await page.keyboard.press("Tab");
    
    await expect(
```

File: src/components/vertical-menu/vertical-menu.pw.tsx [hunk 6]
```diff
@@ -0,0 +1,702 @@
...
+  test(`should navigate to the 4th item using Shift Tab key`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(<VerticalMenuDefaultComponent />);
+
+    await verticalMenuItem(page).nth(5).focus();
+    await page.keyboard.press("Shift+Tab");
+    await page.keyboard.press("Shift+Tab");
+    await expect(
```
Comment (DipperTheDan) [inline src/components/vertical-menu/vertical-menu.pw.tsx:404]:
```suggestion
    await page.keyboard.press("Shift+Tab");
    
    await expect(
```

File: src/components/vertical-menu/vertical-menu.pw.tsx [hunk 7]
```diff
@@ -0,0 +1,702 @@
...
+      verticalMenuItem(page).filter({ hasText: "Item 4" })
+    ).toBeFocused();
+  });
+
+  test(`should expand to item 3 using click`, async ({ mount, page }) => {
+    await mount(<VerticalMenuDefaultComponent />);
+
+    await verticalMenuItem(page).nth(2).click();
+    await expect(
```
Comment (DipperTheDan) [inline src/components/vertical-menu/vertical-menu.pw.tsx:413]:
```suggestion
    await verticalMenuItem(page).nth(2).click();
    
    await expect(
```

File: src/components/vertical-menu/vertical-menu.pw.tsx [hunk 8]
```diff
@@ -0,0 +1,702 @@
...
+      verticalMenuItem(page).locator("h3").filter({ hasText: "Active Item" })
+    ).toBeVisible();
+  });
+
+  test(`should collapse to item 3 using click`, async ({ mount, page }) => {
+    await mount(<VerticalMenuDefaultComponent />);
+
+    await verticalMenuItem(page).nth(2).click();
+    await verticalMenuItem(page).nth(2).click();
+    await expect(
```
Comment (DipperTheDan) [inline src/components/vertical-menu/vertical-menu.pw.tsx:423]:
```suggestion
    await verticalMenuItem(page).nth(2).click();
    
    await expect(
```

File: src/components/vertical-menu/vertical-menu.pw.tsx [hunk 9]
```diff
@@ -0,0 +1,702 @@
...
+  }) => {
+    await mount(<VerticalMenuDefaultComponent />);
+
+    await verticalMenuItem(page).first().focus();
+    await page.keyboard.press("Tab");
+    await page.keyboard.press("Tab");
+    await page.keyboard.press("Enter");
+    await page.keyboard.press("Tab");
+    await page.keyboard.press("Tab");
+    await expect(
```
Comment (DipperTheDan) [inline src/components/vertical-menu/vertical-menu.pw.tsx:476]:
```suggestion
    await page.keyboard.press("Tab");
    
    await expect(
```

File: src/components/vertical-menu/vertical-menu.pw.tsx [hunk 10]
```diff
@@ -0,0 +1,702 @@
...
+  test(`should check the focus styling, focusRedesignOptOut false`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(<VerticalMenuDefaultComponent />);
+
+    await verticalMenuItem(page).first().focus();
+    await page.keyboard.press("Tab");
+    const menuItem2 = verticalMenuItem(page).nth(1);
+    await expect(menuItem2).toHaveCSS(
```
Comment (DipperTheDan) [inline src/components/vertical-menu/vertical-menu.pw.tsx:503]:
```suggestion
    const menuItem2 = verticalMenuItem(page).nth(1);
    
    await expect(menuItem2).toHaveCSS(
```

File: src/components/vertical-menu/vertical-menu.pw.tsx [hunk 11]
```diff
@@ -0,0 +1,702 @@
...
+      mount,
+      page,
+    }) => {
+      await mount(<VerticalMenuFullScreenBackgroundScrollTest />);
+
+      await page.keyboard.press("Tab");
+      await page.keyboard.press("Tab");
+      await page.keyboard.press("Tab");
+      await page.keyboard.press("Tab");
+      await expect(closeIconButton(page)).toBeFocused();
```
Comment (DipperTheDan) [inline src/components/vertical-menu/vertical-menu.pw.tsx:524]:
```suggestion
      await page.keyboard.press("Tab");
      
      await expect(closeIconButton(page)).toBeFocused();
```

File: src/components/vertical-menu/vertical-menu.pw.tsx [hunk 12]
```diff
@@ -0,0 +1,702 @@
...
+    test(`tabbing backward through the menu and back to the start should not make the background scroll to the bottom`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<VerticalMenuFullScreenBackgroundScrollTest />);
+
+      await page.keyboard.press("Shift+Tab");
+      await page.keyboard.press("Shift+Tab");
+      await page.keyboard.press("Shift+Tab");
+      await expect(closeIconButton(page)).toBeFocused();
```
Comment (DipperTheDan) [inline src/components/vertical-menu/vertical-menu.pw.tsx:537]:
```suggestion
      await page.keyboard.press("Shift+Tab");
      
      await expect(closeIconButton(page)).toBeFocused();
```

File: src/components/vertical-menu/vertical-menu.pw.tsx [hunk 13]
```diff
@@ -0,0 +1,702 @@
...
+      <VerticalMenuTriggerCustom
+        onClick={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+
+    await verticalMenuTrigger(page).click();
+
+    expect(callbackCount).toBe(1);
```
Comment (DipperTheDan) [inline src/components/vertical-menu/vertical-menu.pw.tsx:559]: suggestion/question(non-blocking): same for the others too. I thought you'd need an `await` before this?
```suggestion
    await expect(callbackCount).toBe(1);
```

File: src/components/vertical-menu/vertical-menu.pw.tsx [hunk 14]
```diff
@@ -0,0 +1,702 @@
...
+  });
+
+  test(`should pass accessibility tests when open`, async ({ mount, page }) => {
+    await mount(<VerticalMenuDefaultComponent />);
+
+    await verticalMenuItem(page).first().focus();
+    await page.keyboard.press("Tab");
+    await page.keyboard.press("Tab");
+    await page.keyboard.press("Enter");
+    await checkAccessibility(page);
```
Comment (DipperTheDan) [inline src/components/vertical-menu/vertical-menu.pw.tsx:629]:
```suggestion
    await page.keyboard.press("Enter");
    
    await checkAccessibility(page);
```

File: src/components/vertical-menu/components.test-pw.tsx (hunk 2 above)
Comment (DipperTheDan) [reply src/components/vertical-menu/components.test-pw.tsx:102]: Spoke about this privately and they need to stay.

File: src/components/vertical-menu/components.test-pw.tsx [hunk 15]
```diff
@@ -0,0 +1,382 @@
...
+          />
+          <VerticalMenuItem href="/child-item-2" title="ChildItem 2" />
+        </VerticalMenuItem>
+      </VerticalMenu>
+    </Box>
+  );
+};
+
+export const VerticalMenuItemCustomHref = ({
+  component,
```
[the remark this answers is not in the document]
Comment (DipperTheDan) [reply src/components/vertical-menu/components.test-pw.tsx:138]: Spoke about this privately and they need to stay.

**Review by divyajindel — APPROVED** [review]

**Pull request merged**